services/frontend-service/src/ui/AppBar/ActionsCart.tsx

```diff
@@ -70,6 +70,17 @@ const ActionsList = (props: { openNotification: (msg: string) => void }) => {
     }, [openNotification]);
     const [doActions, doActionsState] = callbacks.useBatch(actions, actionsSucceeded, actionsFailed);
 
+    useEffect(() => {
+        if (actions.length) {
+            window.onbeforeunload = (e: BeforeUnloadEvent) => {
```

Comment:
This onbeforeunload never gets removed from the window. Does that mean that once you have an action list you cannot leave anymore?
This library does it correct btw: https://www.npmjs.com/package/react-beforeunload